### PR TITLE
Merge build and deploy docs jobs for GitHub CI

### PR DESCRIPTION
**Pull request by charlesbvll**
### Description

The docs workflow is redundant because we build the docs 2 times for the main branch.

### Related issues/PRs

N/A

### Explanation

Remove the deploy job and replace it by a conditional step at the end of the build job.

### Checklist

- [x] Implement proposed change
- [x] Make CI checks pass
- [x] Ping maintainers on [Slack](https://flower.dev/join-slack/) (channel `#contributions`)

### Any other comments?

N/A

### CODE REVIEW

**Comment by tanertopal** [comment]
@charlesbvll I realize we invoke `make docs` in `deploy-docs.sh` and invoke `make html` in `build-docs.sh`. Do you know what the impact would be?

Also should we delete the `deploy-docs.sh` script with this change?

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ env:
 jobs:
   build_docs:
```
Comment (tanertopal) [inline .github/workflows/docs.yml:15]:
```suggestion
  build_and_deploy:
```

**Comment by tanertopal** [comment]
@charlesbvll should we delete the unused `deploy-docs.sh` script?

**Comment by charlesbvll** [comment]
> @charlesbvll should we delete the unused `deploy-docs.sh` script?

I think so, yes. I just wanted to first merge this PR in order to make sure that everything was working properly before removing it. But if you prefer I can remove it in this PR

**Comment by tanertopal** [comment]
> > @charlesbvll should we delete the unused `deploy-docs.sh` script?
> 
> I think so, yes. I just wanted to first merge this PR in order to make sure that everything was working properly before removing it. But if you prefer I can remove it in this PR

Sounds reasonable. Merging it.

**Review by tanertopal — APPROVED** [review]

**Review by tanertopal — APPROVED** [review]

**Pull request merged**